Mathlib/Algebra/DualNumber.lean

```diff
@@ -95,40 +95,88 @@ theorem commute_eps_left [Semiring R] (x : DualNumber R) : Commute ε x := by
 /-- `ε` commutes with every element of the algebra. -/
 theorem commute_eps_right [Semiring R] (x : DualNumber R) : Commute x ε := (commute_eps_left x).symm
 
-/-- For two algebra morphisms out of `R[ε]` to agree, it suffices for them to agree on `ε`. -/
+/-- For two algebra morphisms out of `A[ε]` to agree, it suffices for them to agree on the elements
+of `A` and the `A`-multiples of `ε`. -/
 @[ext]
-theorem algHom_ext {A} [CommSemiring R] [Semiring A] [Algebra R A] ⦃f g : R[ε] →ₐ[R] A⦄
-    (h : f ε = g ε) : f = g :=
-  algHom_ext' <| LinearMap.ext_ring <| h
+theorem algHom_ext {A} [CommSemiring R] [Semiring A] [Semiring B] [Algebra R A] [Algebra R B]
```

Comment:
I think there is value in keeping the old version of ext. It is a special case, but the side conditions are a lot simpler.

Reply:
That seems fair, I've added it back. Ideally it would follow from `ext <;> simp [*]`, but apparently some results needed to make that work are in un-imported files.

Reply:
Ah, it now works with a similar proof!